app/controllers/stocks_controller.rb

```diff
@@ -10,6 +10,16 @@ def update
     redirect_back_or_to @stock.product
   end
 
+  def edit_transfer
+    @product = Product.find(params[:id])
+  end
+
+  def transfer
+    @product = Product.find(params[:id])
```

Comment:
We don't need to use an instance variable here. In edit_transfer @ is needed because we are passing the product to a view. Here regular variable is enough :)